adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/filter/KeycloakPreAuthActionsFilter.java

```diff
@@ -93,4 +95,23 @@ public void setUserSessionManagement(UserSessionManagement userSessionManagement
     public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
         this.applicationContext = applicationContext;
     }
+    
+    void setNodesRegistrationManagement(NodesRegistrationManagement nodesRegistrationManagement) {
+        this.nodesRegistrationManagement = nodesRegistrationManagement;
+    }
+    
+    void setPreAuthActionsHandlerFactory(PreAuthActionsHandlerFactory preAuthActionsHandlerFactory) {
+        this.preAuthActionsHandlerFactory = preAuthActionsHandlerFactory;
+    }
+    
+    /**
+     * Creates {@link PreAuthActionsHandler}s.
+     * 
+     * Package-private class to enable mocking.
+     */
+    class PreAuthActionsHandlerFactory {
```

Comment:
would have liked to use `java.util.function.Function` but adapters have source/target level 1.6 :(

Reply:
Did the factory make testing easier?

Reply:
Without the factory unit testing (mocking) would really be possible, without testing the pre auth action handler at the same time.